twine/exceptions.py

```diff
@@ -29,7 +29,16 @@ class RedirectDetected(TwineException):
     redirecting them.
     """
 
-    pass
+    @classmethod
+    def from_args(cls, repository_url, redirect_url):
+        msg = "\n".join([
+            "Unexpected redirect from {0} to {1}."
```

Comment:
I don't think we support 2.6 any longer meaning you can drop the numbers here, if you'd like.